code/src/lidar_distance_sensor_c.cpp

```diff
@@ -0,0 +1,24 @@
+#include "lidar_distance_sensor_c.hpp"
+
+namespace r2d2::distance
+{
+    void lidar_distance_sensor_c::fill_distance_frame(frame_distance_s& frame){
+
+        frame.sensor_type = 0; // 0 = lidar, 1 = sonar
+        frame.sensor_pov = 0;
+        frame.count = 45;
+
+        // 16 packets for one 360 degree measurement
+        for (int i = 0; i < 16; i++) {
+            lidar.receive_packet();
+        }
+        
+        // save a measurement every 8 degrees.
+        for (int i = 0; i < 45; i++) {
```

Comment:
I get where the 45 is for but now this is a magic number. can this be in a centralized place?

Reply:
The comment should be enough, might have been 360/8, but clear nonetheless

Reply:
it is bad practice to put hard coded numbers in loops

Reply:
It should have been dynamic int that changes depending on the resolution the user wanted, you are right, but due to a lack of time we can not properly implement it.